### ISSUE TITLE
Tabulator: error raised when editing a value in a client-side filtered column with `show_index=False`

### ISSUE THREAD

**Issue by maximlt** [holoviz/panel#3984]
To reproduce:
```python
import pandas as pd
import panel as pn

df = pd.DataFrame({
        'x': [0, 0, 1],
        'y': ['a', 'b', 'c'],
    },
    index=['i1', 'i2', 'i3'],
)

w = pn.widgets.Tabulator(
    df,
    header_filters=True,
    show_index=False,
).servable()
```

In this line the DataFrame `df` doesn't have an `index` column when `show_index=False`:

https://github.com/holoviz/panel/blob/b83178da8cc0857d7536d331b241c74da527d01d/panel/widgets/tables.py#L1253

**Comment by maximlt** [comment]
I noticed that this piece of code in `Tabulator._process_event` happens to work in - I think - an unexpected way. It is supposed to keep track of the edited indexes, except that `_process_event` is executed any time either a ClickEvent or a TableEditEvent is triggered. Editing a cell value implies first clicking on the cell (in most cases), this leads to a first ClickEvent being triggered which populates the list of edited indexes. It's only once the value is edited that the TableEditEvent is triggered and `_process_event` executed.

Changing this piece of code to store the edited indexes only in the case of an TableEditEvent wouldn't work, as `_process_event` is executed **after** `_process_data`, while `_process_data` requires the updated list of edited indexes to work as expected.

https://github.com/holoviz/panel/blob/b83178da8cc0857d7536d331b241c74da527d01d/panel/widgets/tables.py#L1169-L1173

I think it'd be possible to clean up the current list of maybe edited indexes maintaining another collection of the actual edited indexes.

**Comment by maximlt** [comment]
This issue also arises when the DataFrame index is displayed (show_index=True, its default) but isn't called `'index'` (`df.index.name = 'other name'`).

When the index is unnamed or named `'index'`, a workaround for this issue is to set `show_index=True` and `hidden_columns=['index']`.

**Comment by philippjfr** [comment]
Thanks. Think we can get a fix before end of week? Would like to get 0.14.1 out asap.

**Comment by philippjfr** [comment]
I'll see if I can find a fix.
